### ISSUE TITLE
Referencing outputs by make/model isn't sufficient in general

### ISSUE THREAD

**Issue by ids1024** [pop-os/iced-sctk#1]
`IcedOutput` (currently defined in the `iced_native` fork) allows specifying `All`, `Active`, or an ouput by make/model.

Make/model pairs can't be assumed to be unique in general (a user with multiple monitors may have multiple of the same). And monitor EDID information can't really be trusted in general; even the serial number can't be trusted to be unique.

I guess to refer to the currently attached outputs, it would be necessary to expose wayland-rs/sctk types in the API, or expose opaque wrappers?

For the workspaces view it is necessary to create a surface on each monitor, and it needs to know what `WlOuput` it corresponds to to know what to show. So there will need to be some way for it to associate outputs with their `WlOutput` id.

**Comment by ids1024** [comment]
Hm, I guess to match the output by Wayland object ID, it needs to expose a way for the application to enumerate outputs and detect changes, using the same `WlOutput` instances that are used internally in iced-sctk. If the screen capture code created its own registry and output objects from that, I don't think they can be reliably matched.

**Comment by Drakulix** [comment]
Note that the protocol also explicitly discourages from using `wl_output::make` and `wl_output::model`:
> Note: wl_output only advertises partial information about the output position and identification. Some compositors, for instance those not implementing a desktop-style output layout or those exposing virtual outputs, might fake this information. Instead of using x and y, clients should use xdg_output.logical_position. Instead of using make and model, clients should use name and description.

Additionally `wl_output::name` is also not a good identifier, given that it is not guaranteed to be stable:

> The name is not guaranteed to be persistent across sessions, thus cannot be used to reliably identify an output in e.g. configuration files.

On the other side `xdg_output::name` seems to have many of the qualities we are looking for and might be sufficient:

> Each name is unique among all wl_output globals, but if a wl_output global is destroyed the same name may be reused later. The names will also remain consistent across sessions with the same hardware and software configuration. [...] This event is only sent once per xdg_output, and the name does not change over the lifetime of the wl_output global.

Depending on `xdg-output` support for `iced-sctk` in general might be a little much, but at least for our clients we could use that as a required identifier.

**Comment by ids1024** [comment]
Good point, `xdg_output::name` is probably our best bet for configuration that persists across sessions.

Within one session, I guess WlOutput -> name -> WlOutput mapping could be subject to races, but hopefully you're not disconnecting monitors and connecting different ones while the workspace view is opening. Or I guess there's monitor configuration changes while it's open. I suppose this mapping would at least be "eventually consistent" as long as you stop changing monitors at some point. :laughing:

Edit: Or maybe there's an edge case where it could race if the cosmic-workspaces code tries to create a surface before iced-sctk knows the output has been replaced with one with the same name. Not a very likely issue.
